db.go

```diff
@@ -31,10 +32,40 @@ import (
 	"github.com/percona/qan-api2/migrations"
 )
 
+const (
+	databaseNotExist = 81
+)
+
 // NewDB return updated db.
 func NewDB(dsn string, maxIdleConns, maxOpenConns int) *sqlx.DB {
 	db, err := sqlx.Connect("clickhouse", dsn)
 	if err != nil {
+		if exception, ok := err.(*clickhouse.Exception); ok {
+			if exception.Code == databaseNotExist {
+				log.Println("Creating database")
+				clickhouseURL, err := url.Parse(dsn)
+				if err != nil {
+					log.Fatal(err)
+				}
+				q := clickhouseURL.Query()
+				databaseName := q.Get("database")
+				q.Set("database", "default")
+
+				clickhouseURL.RawQuery = q.Encode()
+
+				defaultDB, err := sqlx.Connect("clickhouse", clickhouseURL.String())
+				if err != nil {
+					log.Fatal("Connection: ", err)
+				}
+				result, err := defaultDB.Exec(fmt.Sprintf(`CREATE DATABASE %s ENGINE = Ordinary`, databaseName))
+				if err != nil {
+					log.Fatalf("Create database. Result: %v, Error: %v", result, err)
+				}
+				log.Println("Database was created. Now we should restart qan-api2")
```

Comment:
Why do we should restart qan-api2? Can't we proceed? reconnect to db here and put log.Fatal into `else`?

Reply:
We can but I wanted to use the same flow for every connection. If we'll open a connection here we'll have two slightly different workflows. Also, we need to close the old connection and so on.